### PR TITLE
fixes #2248 adds api controllers and actions to access_permissions.rb

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: app/controllers/api/v1/home_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module Api
   module V1
     class HomeController < V1::BaseController
+      before_filter :require_admin
```
Comment (ohadlevy) [inline app/controllers/api/v1/home_controller.rb:4]: any reason not to put it in the parent controller?

Comment (isratrade) [inline app/controllers/api/v1/home_controller.rb:4]: @ohadlevy, only a few controllers :require_admin.  In BaseController, there is before_filter :authorize

Comment (ohadlevy) [inline app/controllers/api/v1/home_controller.rb:4]: oh i see, i misunderstood it.. thanks

File: test/lib/foreman/access_permissions_test.rb [hunk 2]
```diff
@@ -36,8 +36,8 @@ class AccessPermissionsTest < ActiveSupport::TestCase
     controller = "#{r.defaults[:controller]}_controller".classify.constantize
     filters    = controller.send(:_process_action_callbacks)
 
-    # Skip controllers that don't require logins (e.g. API)
-    next if filters.select { |f| f.filter == :require_login }.empty?
+    # Skip controllers that don't require logins
+    next if filters.select { |f| (f.filter == :require_login || f.filter == :authorize)}.empty?
```
Comment (domcleal) [inline test/lib/foreman/access_permissions_test.rb]: This line can be removed completely now, which might prevent accidents later.  You'll need to add "apipie/apipies/index" to the list of MAY_SKIP_REQUIRE_LOGIN above.

**Comment by isratrade** [comment]
thanks @domcleal, I committed your suggested changes.

File: lib/api/authorization.rb [hunk 3]
```diff
@@ -22,6 +22,15 @@ def authenticate
       return true
     end
 
+    def is_admin
```
Comment (ohadlevy) [inline lib/api/authorization.rb]: should this rename to is_admin?

**Comment by isratrade** [comment]
@ohadlevy re-commited with is_admin?

**Comment by ohadlevy** [comment]
your commit msg should close http://projects.theforeman.org/issues/2248

**Comment by isratrade** [comment]
@ohadlevy commit message is updated

**Comment by domcleal** [comment]
[test]

**Comment by ohadlevy** [comment]
@isratrade can you rebase & test please?

**Comment by isratrade** [comment]
@ohadlevy rebased and tested

**Comment by isratrade** [comment]
@ohadlevy rebased

File: app/controllers/api/v1/home_controller.rb (hunk 1 above)
Comment (ohadlevy) [inline app/controllers/api/v1/home_controller.rb:4]: hmm.. why do we force every action in the home controller to require admin?

**Comment by isratrade** [comment]
@ohadlevy rebased

**Comment by ohadlevy** [comment]
merged - thanks

**Pull request closed without merging**